Allow passing multiple attrs args into components

gazpachoking opened the pull request:
Allows passing more than one dict of arguments into an element. Can be useful if you have a helper function that generates a single attribute. This would let you use that helper more than once on the same element.

gazpachoking commented on htpy/__init__.py:
```diff
@@ -265,34 +265,28 @@ def __str__(self) -> _Markup:
-            id_class, attrs = args
+    def __call__(self: BaseElementSelf, /, *args: t.Any, **kwargs: t.Any) -> BaseElementSelf:
+        id_class: str = ""
+        attr_dicts: list[Mapping[str, Attribute]]
+        attrs: dict[str, Attribute] = {}
+
+        if args and not isinstance(args[0], Mapping):
+            id_class, *attr_dicts = args
+        else:
+            attr_dicts = list(args)
```
That was to make the type checker happy. I could just declare it as an iterable though.

gazpachoking replied:
Oh, lol. I failed to even see your suggested change doing exactly that until now! 😅

pelme replied:
Haha, no worries!

pelme commented:
Someone else was running into similar problems:

https://www.reddit.com/r/htmx/comments/1kantse/comment/mpxh5jv/

I really think we need to improve the docs here :)

gazpachoking commented:
Okay, fixed up that list call and added a tiny bit of docs.

> Someone else was running into similar problems:
> 
> https://www.reddit.com/r/htmx/comments/1kantse/comment/mpxh5jv/
> 
> I really think we need to improve the docs here :)

I didn't quite get their need, unless they were over complicating it. Was it just working around not being able to pass in multiple attribute dicts?

pelme commented:
> Okay, fixed up that list call and added a tiny bit of docs.
> 
> > Someone else was running into similar problems:
> > https://www.reddit.com/r/htmx/comments/1kantse/comment/mpxh5jv/
> > I really think we need to improve the docs here :)
> 
> I didn't quite get their need, unless they were over complicating it. Was it just working around not being able to pass in multiple attribute dicts?

Yes I think it looks a bit over complicated. Being able to pass multiple dicts of attributes seems like an easy way to combine them and your docs example show how.

Hopefully we can get a community going of helpers that generate attribute dicts that can then be combined :)
